### DISCUSSION TITLE
v.0.4.0-beta

### DISCUSSION THREAD

**Discussion by asim-shrestha** [reworkd/AgentGPT#540] [Announcements]
<h1 align="center">
🎉🤖&nbsp;&nbsp;&nbsp;Beta 4.0&nbsp;&nbsp;&nbsp;🤖🎉
</h1>

Welcome to the **Beta 4.0** release of AgentGPT! With this comes a slew of features, here are the most prevalent: 
- The reintroduction of a basic web search. We've also added a daily usage limit to accounts so we can keep costs under control
- The beginning of the migration to a Python backend
  - Python allows us to better integrate with the python AI ecosystem
  - Within Python we're also developing a slew of tools that agents can interact with such as web search, web browsing, wikipedia, and google productivity tools
  - The Python model itself also has improved handling for runs, currently this is only available locally for developers  
- Misc UI improvements

This next week will focus on the following
- Getting our Python backend up and running
- Subscriptions to avoid usage limits
- Improved Agent handling via the ReAct framework for task execution
- Additional tool support

We will be releasing updates on a weekly schedule, stay tuned every Tuesday for the latest improvements to AgentGPT 🚀 If you'd like to help shape these updates, please fill out this brief [survey](https://forms.gle/fjCo72hiYWZydJa1A) and join our [discord server](https://discord.gg/jdSBAnmdnY). We would love to hear your feedback. ❤️ 

## What's Changed
* 📃 Add examples to the front page and fix prompts by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/437
* 🌐 i18n updates and fixes | fonts to export PDF by @Cs4K1Sr4C in https://github.com/reworkd/AgentGPT/pull/446
* 🐍 Add backend python platform by @awtkns in https://github.com/reworkd/AgentGPT/pull/451
* 🚫 removed @types/micro by @souravghosh01 in https://github.com/reworkd/AgentGPT/pull/457
* 🚫 Delete tasks by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/458
* ✨ Add tasks by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/460
* 🔐 Local Auth Improvements by @awtkns in https://github.com/reworkd/AgentGPT/pull/449
* 🔨 PDF export overlapping by @Cs4K1Sr4C in https://github.com/reworkd/AgentGPT/pull/465
* 🔨 move mode toggle out of settings by @Jshen123 in https://github.com/reworkd/AgentGPT/pull/469
* 🌐 Add Traditional Chinese Translation + edit Simplified Chinese missing translation  by @enjuichang in https://github.com/reworkd/AgentGPT/pull/482
* 🔨 PDF export by @Cs4K1Sr4C in https://github.com/reworkd/AgentGPT/pull/470
* Api Key Toggle View by @rp-bot in https://github.com/reworkd/AgentGPT/pull/489
* 📄 Add buttons to display TaskWindow on mobile by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/485
* ✈️ Plane by @awtkns in https://github.com/reworkd/AgentGPT/pull/488
* ♻ Monorepo by @awtkns in https://github.com/reworkd/AgentGPT/pull/491
* 🐋 Put MySQL DB in docker by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/492
* ✨ Mono documentation by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/493
* ✨ Mono python migration Main by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/498
* 📚 Demonstrate wiki search and tools by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/496
* ✨ clarify difference to chatgpt by @nczempin in https://github.com/reworkd/AgentGPT/pull/500
* 🐛 Fix issues with docker compose by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/502
* 🔥 Hot reload for frontend by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/503
* 🖥 Enable the ability to switch APIs by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/504
* 🔨 Update common.json by @angeousta in https://github.com/reworkd/AgentGPT/pull/505
* 🔨 Fix Invalid environment variables(NEXT_PUBLIC_BACKEND_URL: Required) by @SnoopyDevelops in https://github.com/reworkd/AgentGPT/pull/509
* :memo: Update README.md by @awtkns in https://github.com/reworkd/AgentGPT/pull/512
* 🔨 Fixed Dockerfile and entrypoint.sh to handle Windows line endings by @lauralex in https://github.com/reworkd/AgentGPT/pull/514
* 🤖 Update Sponsors by @reworkd-admin in https://github.com/reworkd/AgentGPT/pull/517
* ✨ Modify Manual Setup by @thame in https://github.com/reworkd/AgentGPT/pull/518
* ⏰ Entrypoint that waits for DB in Python BE by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/516
* 🐍 Python tools by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/519
* 🧼 Cleanup Code by @awtkns in https://github.com/reworkd/AgentGPT/pull/522
* 🐍 Python serper by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/523
* 📚 Update documentation by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/525
* 🤖 Update Sponsors by @reworkd-admin in https://github.com/reworkd/AgentGPT/pull/524
* 💬 Update start agent prompt by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/526
* 💬 Fix remove tasks prefix by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/528
* 🤖 Update Sponsors by @reworkd-admin in https://github.com/reworkd/AgentGPT/pull/529
* 💬 Remove leading dot space by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/530
* 🚫 Remove client side by @asim-shrestha in https://github.com/reworkd/AgentGPT/pull/531
* 🐍Use python as a backend by @awtkns in https://github.com/reworkd/AgentGPT/pull/535

## New Contributors
* @souravghosh01 made their first contribution in https://github.com/reworkd/AgentGPT/pull/457
* @enjuichang made their first contribution in https://github.com/reworkd/AgentGPT/pull/482
* @nczempin made their first contribution in https://github.com/reworkd/AgentGPT/pull/500
* @angeousta made their first contribution in https://github.com/reworkd/AgentGPT/pull/505
* @SnoopyDevelops made their first contribution in https://github.com/reworkd/AgentGPT/pull/509
* @lauralex made their first contribution in https://github.com/reworkd/AgentGPT/pull/514
* @thame made their first contribution in https://github.com/reworkd/AgentGPT/pull/518

**Full Changelog**: https://github.com/reworkd/AgentGPT/compare/v.0.3.0-beta...v.0.4.0-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/reworkd/AgentGPT/releases/tag/v.0.4.0-beta'>v.0.4.0-beta</a>.</em>

**Comment by The0ne23** [comment]
why cant I websearch on local instance of agent gpt

**Comment by ratbuss** [comment]
what's the problem of 0.4 beta run, under conda env :
`E:\.AIGC\AgentGPT\next>npm run dev

> agent-gpt@0.2.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
❌ Invalid environment variables:
 NEXT_PUBLIC_BACKEND_URL: Required

file:///E:/.AIGC/AgentGPT/next/src/env/client.mjs:22
  throw new Error("Invalid environment variables");
        ^

Error: Invalid environment variables
    at file:///E:/.AIGC/AgentGPT/next/src/env/client.mjs:22:9
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.16.0`
